src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -0,0 +1,206 @@
+            {
+                unchecked
+                {
+                    metricsEventDescriptor = new EventDescriptor(0xe232, 0x0, 0x10, 0x4, 0x0, 0x0, (long)0x8000000000000000);
+                }
+            }
+
+            internal FunctionActivityTracker()
+            {
+                Task.Run(
+                    async () =>
+                    {
+                        try
+                        {
+                            while (!etwTaskcancellationSource.Token.IsCancellationRequested)
+                            {
+                                RaiseMetricsEtwEvent(ExecutionStage.InProgress);
+                                await Task.Delay(TimeSpan.FromSeconds(5), etwTaskcancellationSource.Token);
+                            }
+                        }
+                        catch (TaskCanceledException)
+                        {
+                            // This exception gets throws when cancellation request is raised via cancellation token.
+                            // Let's eat this exception and continue
+                        }
+                    },
+                    etwTaskcancellationSource.Token);
+            }
+
+            private enum ExecutionStage
+            {
+                InProgress,
+                Finished
+            }
+
+            internal bool IsActive
+            {
+                get
+                {
+                    return runningFunctionCount != 0;
+                }
+            }
+
+            protected virtual void Dispose(bool disposing)
+            {
+                if (disposing)
+                {                    
+                    etwTaskcancellationSource.Dispose();
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+
+           internal void BeginFunctionEventHandler()
+           {
+                totalExecutionCount++;
+                runningFunctionCount++;
+           }
+
+            internal void EndFunctionEventHandler()
+            {
+                if (runningFunctionCount > 0)
+                {
+                    runningFunctionCount--;
+                }
+            }
+
+            internal void StopEtwTaskAndRaiseFinishedEvent()
+            {
+                etwTaskcancellationSource.Cancel();
+                RaiseMetricsEtwEvent(ExecutionStage.Finished);
+            }
+
+            private void RaiseMetricsEtwEvent(ExecutionStage executionStage)
+            {
+                var timeSpan = (ulong)(DateTime.UtcNow - startTime).TotalMilliseconds;
+                var executionCount = totalExecutionCount;
+                WriteFunctionsMetricsEvent(executionId, timeSpan, executionCount, executionStage.ToString());
+            }
+
+            private static bool WriteFunctionsMetricsEvent(string executionId, ulong executionTimeSpan, ulong executionCount, string executionStage)
```

Comment:
This function returns bool, but nobody consumes the result. Either use or remove the value?